lib/membrane_rtc_engine/engine.ex

```diff
@@ -341,6 +351,51 @@ defmodule Membrane.RTC.Engine do
     {actions, state}
   end
 
+  defp handle_media_event(
+         %{type: :update_peer_metadata, data: %{metadata: metadata}},
+         peer_id,
+         _ctx,
+         state
+       ) do
+    peer = Map.get(state.peers, peer_id)
+
+    if peer.metadata != metadata do
+      peer = %{peer | metadata: metadata}
+      state = put_in(state, [:peers, peer_id], peer)
+
+      MediaEvent.create_peer_updated_event(peer_id, peer)
+      |> dispatch()
+
+      {[], state}
+    else
+      {[], state}
+    end
```

Comment:
You can delete else as we always return `{[], state}`

Reply:
Yes but in if we update state, so I can change it like this
```elixir
...
state = if ... do
   ...
   state
else
   state
end
{[],state}
```
But I am not sure if this looks better